### ISSUE TITLE
Add message with optimize route result

### ISSUE THREAD

**Issue by stevejgordon** [HTBox/allReady#1578]
We should cache and display any errors from the optimize route service so that the user is aware of the failure. Currently this fails silently.

We could cache an error using the user id and on the view model for the itinerary details page, check this cache if any new messages exist which need to be displayed.

**Comment by chinwobble** [comment]
Is the intended use for caching the result for a scenario such as:
1. the database has bad addresses /coordinates for the waypoints
2. the optimise route criteria with the bad addresses / coordinates is sent to google which is unable to resolve a route 
3. The results are cached so that the UI does not repeatedly attempt to optmise and unresolvable path
4. Users or admin should be prompted  somehow and check the addresses are correct

**Comment by stevejgordon** [comment]
@chinwobble - II'd forgotten I already logged this when I put in #1709! In this case it's more around the flow as I've just commented on #1709. So we cache the result of the optimization so that in the UI we can advise the user of the that result. Rather than loading the page and giving no information back to the end user.

We now have code which geocodes an address to lat/long coordinates when we save it, so I'm hoping we shouldn't see this often. In theory we look to capture and reject any requests for getting into our system when they have bad addresses.
